src/Compilers/CSharp/Portable/Symbols/TypeSymbol.cs

```diff
@@ -47,6 +47,8 @@ private class InterfaceInfo
             // all directly implemented interfaces, their bases and all interfaces to the bases of the type recursively
             internal ImmutableArray<NamedTypeSymbol> allInterfaces;
 
+            internal ImmutableArray<INamedTypeSymbol> allInterfacesPublicSymbols;
```

Comment:
>internal ImmutableArray<INamedTypeSymbol> allInterfacesPublicSymbols; [](http://example.com/codeflow?start=12&length=69)

I think this cache belongs in public symbol implementation

Reply:
@AlekseyTs -- please take another look and see if this matches your expectations

Reply:
> there was some internal symbol caching at the level where I added it

Could you point to the specific instance? In case we would want to adjust it.

Reply:
My understanding is the [InterfaceInfo](https://github.com/dotnet/roslyn/blob/8e3d84569a17786744edc314614a99800ec3fd1f/src/Compilers/CSharp/Portable/Symbols/TypeSymbol.cs#L45) is what provides the cached data for the TypeSymbol, and that's where I had hooked into before.